### PR TITLE
ucs_storage_profile module and integration tests

### PR DESCRIPTION
**Pull request by dsoper2**
##### SUMMARY

Configures storage profiles on Cisco UCS Manager.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ucs_storage_profile
##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (storage_prof 74c94e0038) last updated 2018/07/02 12:50:52 (GMT -500)
  config file = /Users/dsoper/.ansible.cfg
  configured module search path = ['/Users/dsoper/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dsoper/Documents/ansible2/lib/ansible
  executable location = /Users/dsoper/Documents/ansible2/bin/ansible
  python version = 3.6.5 (default, Apr 20 2018, 18:22:17) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by dsoper2** [comment]
shipit

File: lib/ansible/modules/remote_management/ucs/ucs_storage_profile.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+from ansible.module_utils.remote_management.ucs import UCSModule, ucs_argument_spec
+
+
+def main():
+    argument_spec = ucs_argument_spec
+    argument_spec.update(
+        org_dn=dict(type='str', default='org-root'),
+        name=dict(type='str', required=True),
+        description=dict(type='str', aliases=['descr'], default=''),
+        local_luns=dict(type='list'),
```
Comment (dsoper2) [inline lib/ansible/modules/remote_management/ucs/ucs_storage_profile.py]: Use nested argument_spec to describe list for local_luns

Comment (dagwieers) [inline lib/ansible/modules/remote_management/ucs/ucs_storage_profile.py]: I think you can benefit from a nested argument_spec here.

For the module documentation you can use the **suboptions** entry, for the argument_spec you can add an **options** key.

One simple example showing how it works is the iptables module.
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/iptables.py#L541-L545

A more extensive example which is closer to your needs here is the meraki_mx_l3_firewall module.
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/meraki/meraki_mx_l3_firewall.py#L238

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/remote_management/ucs/ucs_storage_profile.py (hunk 1 above)
Comment (dsoper2) [reply lib/ansible/modules/remote_management/ucs/ucs_storage_profile.py]: Suboptions and nested argument spec now added.

**Review by dagwieers — APPROVED** [review]

**Comment by dagwieers** [comment]
I am going to merge this, so we can get feedback sooner rather than later, before v2.7 freeze nears.

One small note regarding the use of strings for integer values. I don't know why this is done and we don't support integers for integer values here.

**Pull request merged**